assign_rights/assemble.py

```diff
@@ -52,9 +55,15 @@ def calculate_dates(self, object, request_start_date, request_end_date):
             object, "end_date", request_end_date, object.end_date_period)
         return object_start, object_end
 
-    def create_json(self):
+    def create_json(self, obj, object_start, object_end):
         """docstring for create_json"""
-    pass
+        if obj.__class__.__name__ == "RightsShell":
+            serializer = RightsShellSerializer(obj, context={"start_date": object_start, "end_date": object_end})
+        else:
+            serializer = RightsGrantedSerializer(obj, context={"start_date": object_start, "end_date": object_end})
```

Comment:
I would consider passing the serializer class as an argument to this function. That will save you a couple of lines and make this function more flexible.